packages/atoms/src/Select.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { TouchableOpacity } from 'react-native';
+
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { FlagIconName, Icon } from '@trezor/icons';
+
+import { Box } from './Box';
+import { Text } from './Text';
+import { BottomModal } from './Modal/BottomModal';
+import { SelectItem, SelectItemProps } from './SelectItem';
+
+export type SelectItem = {
+    label: string;
+    value: string;
+    iconName: FlagIconName;
+};
+
+type SelectProps = {
+    items: SelectItem[];
+    label: string;
+};
+
+const selectStyle = prepareNativeStyle(utils => ({
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    backgroundColor: utils.colors.gray200,
+    borderWidth: utils.borders.widths.small,
+    borderRadius: utils.borders.radii.small,
+    borderColor: utils.colors.gray200,
+    color: utils.colors.gray600,
+    paddingLeft: 12,
+    paddingRight: 23.25,
+    height: 58,
+}));
+
+export const Select = ({ items, label }: SelectProps) => {
+    const { applyStyle } = useNativeStyles();
+    const [isOpen, setIsOpen] = useState<boolean>(false);
+    const [selectedItem, setSelectedItem] = useState<null | SelectItemProps['value']>(null);
+
+    const handleSelectItem = (value: string) => {
+        setSelectedItem(value);
+        setIsOpen(false);
+    };
+
+    const getLabel = (): string => {
+        if (!selectedItem) return label;
+        return items.find(item => item.value === selectedItem)?.label ?? label;
+    };
+
+    return (
+        <>
+            {isOpen && (
+                <BottomModal
+                    isVisible={isOpen}
+                    onVisibilityChange={setIsOpen}
```

Comment:
Is there a reason to have this done in two different ways? I mean it does the same thing apparently and it could be parametrized and called with some `handleChangeVisibility = (visible: boolean)` fnc. But maybe I am missing something.

```
onVisibilityChange={setIsOpen}
onBackArrowClick={() => setIsOpen(false)}
```

Reply:
Could you please elaborate what specifically is not right? I am not sure I understand what you mean. @bouchja1